src/jit/scopeinfo.cpp

```diff
@@ -947,6 +948,7 @@ void                CodeGen::psiBegProlog()
 
                     newScope->scRegister = true;
                     newScope->u1.scRegNum = (regNumberSmall)regNum;
```

Comment:
It appears that you will overwrite scRegNum with REG_NA the second time through this loop, unless I am misunderstanding something. I think that you want to move the declaration of regnum and otherRegNum outside the loop, as well as the assignment to the fields of newScope.